### PR TITLE
Fix homepage and environment setup docs styling

### PR DESCRIPTION
**Pull request by jonsamp**
# Why

Addresses #4124.

After releasing the home page, I got a few points of feedback to address. This PR:

- Adds dark mode colors to the "Continue with Expo" button on /docs/environment-setup. Pictured below is the hovered state:
  Screenshot 2024-05-20 at 10 02 28 PM
- Removes large format devices when viewing the site on a more narrow page:
  - Wide page shows all of the devices in the illustration
  Screenshot 2024-05-20 at 5 03 36 PM
  - Narrow page shows only Android and iOS devices
  Screenshot 2024-05-20 at 5 03 31 PM
- Added a new illustration for the Framework > Tools card. Previously it looked like Expo was offering tools that are provided by React Native. I updated it to show a pseudo-UI of [Expo Orbit](https://expo.dev/orbit). I tried to make the image timeless (no mentions of the Pixel 8 or iPhone 15) since those will be dated quickly.
  Screenshot 2024-05-20 at 9 57 32 PM
  Screenshot 2024-05-20 at 10 02 13 PM
- Added a more recent video to the Watch section. Now it shows the following video, which is from React Conf 2024: https://www.youtube.com/live/0ckOUBiuxVY?si=PmlrPytEh9V_gDJw&t=2315

# Test plan

Look at the home page and make sure that sections mentioned above look correct. Also click on "Get started" to go to /docs/environment-setup and make sure that the "Continue with Expo" button has the correct colors in light and dark mode.

### CODE REVIEW

**Review by Simek — APPROVED** [review]
Thanks for the updates, LGTM! 👍

File: website/src/pages/Home/Watch/index.js [hunk 1]
```diff
@@ -31,16 +31,16 @@ function Watch() {
       <div className={styles.videos}>
         <div className={styles.videoContainer}>
           <iframe
-            src="https://www.youtube.com/embed/NCAY0HIfrwc"
-            title="Mobile Innovation with React Native, ComponentKit, and Litho"
+            src="https://www.youtube.com/embed/0ckOUBiuxVY?si=1WDgoPjhIGzeLNee&amp;start=2315"
```
Comment (hezi) [inline website/src/pages/Home/Watch/index.js:34]: I'm getting this error for the new video link :( 
Screenshot 2024-05-27 at 10 22 56

Comment (hezi) [reply website/src/pages/Home/Watch/index.js:34]: Ignore - works on the site preview.

File: website/src/pages/Home/Watch/index.js [hunk 2]
```diff
@@ -31,16 +31,16 @@ function Watch() {
...
             frameBorder="0"
             allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
             allowFullScreen
             className={styles.video}
           />
           <div className={styles.videoInfo}>
-            <h4>FB 2019: Mobile innovation with React Native</h4>
-            <p>45:29</p>
+            <h4>React Conf 2024: React Native</h4>
+            <p>55:00</p>
```
Comment (Simek) [inline website/src/pages/Home/Watch/index.js:43]: Let's revert the video change in thich PR, so we can shi[ the fixes, and let's wait for the copped out video of React Native keynote only.

**Review by Simek — APPROVED** [review]

**Pull request merged**